### ISSUE TITLE
Allow setters on DTOs or implement an interface for unit testing

### ISSUE THREAD

**Issue by rcollette** [okta/okta-sdk-dotnet#681]
### Describe the feature request?

Classes such as ActivationToken do not have public property setters and therefore cannot be constructed in in unit tests with expected responses.  They also cannot be mocked using Moq because they do not implement an interface.

Another example, in a unit test, a user is created and subsequently activated.   Activation requires a user id, but the user id cannot be set on the User object, so I cannot unit test my process.

### New or Affected Resource(s)

ActivationToken and User at least but there are others.

**Comment by laura-rodriguez** [comment]
Hi @rcollette,

This request has to be coordinated with API teams that provide the OpenAPI specification. In the meantime, I recommend using [WireMock](https://github.com/WireMock-Net/WireMock.Net) to mock the server response. You can see some examples [here](https://github.com/okta/okta-sdk-dotnet/blob/master/src/Okta.Sdk.UnitTest/Api/UsersApiTests.cs#L59).

I hope this helps.

**Comment by rcollette** [comment]
Have the OpenAPI specs been updated by now @laura-rodriguez

**Comment by aniket-okta** [comment]
The private setters are not hand-written — they come from the OpenAPI spec, where these fields are marked `readOnly: true`, and the generator maps that to `{ get; private set; }`. Changing it would mean overriding the generator for ~280 model files and would let callers set fields the API ignores on write.

For unit tests, mock at the HTTP boundary rather than constructing populated models: `src/Okta.Sdk.UnitTest/Api/UsersApiTests.cs` uses WireMock to return a canned JSON payload, which the deserializer then populates through the private setters. That gives you fully-populated models without changing the surface.

Closing as not planned.

**Comment by rcollette** [comment]
> The private setters are not hand-written — they come from the OpenAPI spec, where these fields are marked `readOnly: true`, and the generator maps that to `{ get; private set; }`. Changing it would mean overriding the generator for ~280 model files and would let callers set fields the API ignores on write.
> 
> For unit tests, mock at the HTTP boundary rather than constructing populated models: `src/Okta.Sdk.UnitTest/Api/UsersApiTests.cs` uses WireMock to return a canned JSON payload, which the deserializer then populates through the private setters. That gives you fully-populated models without changing the surface.
> 
> Closing as not planned.

@aniket-okta That's unfortunate.  Wiremock is a much heavier level of mocking, that might be appropriate for testing the SDK itself, but for SDK consumers, the fact that it's making HTTP calls and what HTTP calls are being made should not need to be tested.  The SDK IS the abstraction, that needs to be mocked.

**Comment by aniket-okta** [comment]
Fair point — WireMock is how the SDK's own tests work, and it was the wrong recommendation for a consumer.

You don't need it. The SDK is already the abstraction: every API class has an interface (`IUserApi`, `IGroupApi`, ...), and the paged return type is `IOktaCollectionClient<T> : IAsyncEnumerable<T>`, so `userApi.ListUsers()` can be mocked with Moq or a hand-written fake.

The private setters don't block building the model you return from that mock, because they are the same path the SDK uses for API responses:

```csharp
var user = JsonConvert.DeserializeObject<User>(@"{
  ""id"": ""00uTestId"",
  ""status"": ""ACTIVE"",
  ""created"": ""2026-01-02T03:04:05.000Z"",
  ""profile"": { ""login"": ""ada@example.com"", ""firstName"": ""Ada"" }
}");

// Id = 00uTestId, Status = ACTIVE, Created = 2026-01-02T03:04:05Z, Profile.Login = ada@example.com
```

I checked that against the current SDK, together with a five-line `IOktaCollectionClient<User>` fake for the paging case. No HTTP involved:

```csharp
sealed class FakeCollection : IOktaCollectionClient<User>
{
    private readonly List<User> _items;

    public FakeCollection(List<User> items) => _items = items;

    public async IAsyncEnumerator<User> GetAsyncEnumerator(CancellationToken cancellationToken = default)
    {
        foreach (var item in _items) { yield return item; }
        await Task.CompletedTask;
    }

    public IOktaPagedCollectionEnumerator<User> GetPagedEnumerator(CancellationToken cancellationToken = default)
        => throw new NotImplementedException();
}
```

If that still doesn't cover what you're testing, tell me which type or member it falls down on and I'll reopen.

**Comment by rcollette** [comment]
But the DTOs themselves are not able to be mocked because they do not have interfaces and they have ReadOnly properties (User, ActivationToken)
